src/context/ContextTypes.ts

```diff
@@ -22,29 +22,26 @@
 // These functions will throw an error if the JSON doesn't
 // match the expected interface, even if the JSON is valid.
 
-export interface Context {
-  id?: { [key: string]: string };
-  name?: string;
-  type: string;
-}
-
 export interface Chart {
-  instruments: Instrument[];
+  instruments: InstrumentElement[];
   otherConfig?: { [key: string]: any };
-  range?: TimeRange;
+  range?: TimeRangeObject;
   style?: Style;
   type: string;
-  id?: { [key: string]: string };
+  id?: { [key: string]: any };
   name?: string;
+  [property: string]: any;
 }
 
-export interface Instrument {
-  id: InstrumentID;
+export interface InstrumentElement {
+  id: PurpleId;
+  market?: PurpleMarket;
   type: string;
   name?: string;
+  [property: string]: any;
 }
 
-export interface InstrumentID {
+export interface PurpleId {
```

Comment:
We can't have types like "PurpleId" or "TentacleId" (below).

I had similar issues with QuickType when I first used it - one has to use the settings in such a way so that it accurately names the types after the schema types.

I can't remember exactly what caused this in the past, but there was a way to get around it. It's a pain, I know.

Reply:
We'll need an answer as to how to achieve that as I can't figure out the arguments needed.

I think that the statement:

> We can't have types like "PurpleId" or "TentacleId" (below).

Isn't actually true however. These type names are only generated for the purposes of referencing within the types, externally you would only use the top-level type named after teh schema, e.g. Instrument, and when type checking it will use/compare the content of the type rather than the the name.

However, I would rathr it was generating InstrumentID, rather than PurpleId.

I would also prefer that it was just genearating Instrument and not also InstrumentElement. Not sure why its done that. However, the type checking it will do based on these is correct